src/main/scala/streamer/DataWriter.scala

```diff
@@ -0,0 +1,233 @@
+package streamer
+
+import chisel3._
+import chisel3.util._
+
+// input and output for data writer (data mover in write mode)
+class DataWriterIO(
+    dataWriterTcdmPorts: Int = DataWriterTestParameters.dataWriterTcdmPorts,
+    tcdmDataWidth: Int = DataWriterTestParameters.tcdmDataWidth,
+    unrollingDim: Int = DataWriterTestParameters.unrollingDim,
+    addrWidth: Int = DataWriterTestParameters.addrWidth,
+    fifoWidth: Int = DataWriterTestParameters.fifoWidth
+) extends Bundle {
```

Comment:
Can we not inherit from a common base class?
These parameters are literally the same?
```
class DataReaderIO(
    dataReaderTcdmPorts: Int = DataReaderTestParameters.dataReaderTcdmPorts,
    tcdmDataWidth: Int = DataReaderTestParameters.tcdmDataWidth,
    unrollingDim: Int = DataReaderTestParameters.unrollingDim,
    addrWidth: Int = DataReaderTestParameters.addrWidth,
    fifoWidth: Int = DataReaderTestParameters.fifoWidth
) extends Bundle {
```